staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -651,17 +801,35 @@ func (r *Reflector) relistResourceVersion() string {
 	r.lastSyncResourceVersionMutex.RLock()
 	defer r.lastSyncResourceVersionMutex.RUnlock()
 
+	if r.lastSyncResourceVersion == "" {
```

Comment:
This isn't correct - I mean, we're changing semantics here.

If `isLastSyncResourceVersionUnavailable` is true, then previously we would return "", and now we will return '0" if also lastSyncResourceVersion is empty.

Let's not change the semantics for the old path.

Reply:
[FWIW, I'm not sure why we actually did that, but even if we want to change it, we should do that explicitly in a separate PR and not change it as part of larger complex PR.]

Reply:
doesn't `setIsLastSyncResourceVersionUnavailable` imply that `r. lastSyncResourceVersion` is `!= ""` ? After all we must have provided some RV to the server. 

Anyway I can bring it back to the original state just in case.